### DISCUSSION TITLE
Questions about minio file read file

### DISCUSSION THREAD

**Discussion by Hello-Linux** [minio/minio#19446] [Q&A] [answered]
I would like to have an in-depth look at the underlying principle of minio reading files. minio performs get operations on files, that is, when files are read, will it read parity shards in the disk? Or do both data shards and parity shards read at the same time?

**Comment by klauspost** [comment, marked as answer]
1) Read metadata (and data if inlined).
2) Request from data drives - or if parity is on local disk read that.
3) Reconstruct if needed
4) Return data.

**Reply by Hello-Linux** [reply]
Thank you very much for your patient answer. According to you, minio will read both the data shards and the parity shards of the file when it initiates a get request file, right, instead of the data shards of the read-only file

**Reply by klauspost** [reply]
It will only read enough data to return the object. If it needs to it will also read parity. If a parity shard is local it will prefer that over a remote data shard.

**Reply by Hello-Linux** [reply]
You mean that under normal circumstances, minio will only read the data shards of the file when it initiates a get request. If the file data is damaged, minio will read both date shards and parity shards when it initiates a get request, right?

**Reply by klauspost** [reply]
Something like that. For every "unavailable" data shards we read a parity shard.

**Reply by Hello-Linux** [reply]
Ok, after your patient answer, I understand very clearly, thank you again!

**Reply by Hello-Linux** [reply]
I hope minio can do better and better
